spec/rubocop/cop/rspec/no_expectation_example_spec.rb

```diff
@@ -137,6 +137,39 @@
     end
   end
 
+  context 'when `AllowedPatterns: [^expect_]`' do
+    let(:cop_config) { { 'AllowedPatterns' => ['^expect_'] } }
+
+    context 'with only allowed pattern methods are used' do
+      it 'registers no offenses' do
+        expect_no_offenses(<<~RUBY)
+          it { expect_something }
+        RUBY
+      end
+    end
+
+    context 'with allowed pattern methods and other method are used' do
+      it 'registers no offenses' do
+        expect_no_offenses(<<~RUBY)
+          it do
+            do_something
+            expect_something
```

Comment:
perhaps you can demonstrate the recursiveness of the node_search with nested blocks for instance

Reply:
Tests were added. How do you like it?

Reply:
Thank you :bow: